Project/Source/Components/UI/ComponentTransform2D.cpp

```diff
@@ -202,18 +202,22 @@ const float4x4 ComponentTransform2D::GetGlobalMatrix() const {
 	return globalMatrix;
 }
 
-const float4x4 ComponentTransform2D::GetGlobalMatrixWithSize(bool view3DActive) const {
+const float4x4 ComponentTransform2D::GetGlobalScaledMatrix(bool scaleFactored, bool view3DActive) const {
+	ComponentCanvasRenderer* canvasRenderer = GetOwner().GetComponent<ComponentCanvasRenderer>();
+
+	float factor = scaleFactored ? canvasRenderer ? canvasRenderer->GetCanvasScreenFactor() : 1 : 1;
```

Comment:
It could use some parentheses for a better readability

Reply:
Yup